src/app/modules/workspace/add-improvements/add-improvements.component.html

```diff
@@ -16,14 +17,17 @@
     </button>
   </mat-form-field>
 </div>
+<div *ngIf = "!dataSource" class="load">
```

Comment:
What happens if data is blank?

Reply:
then count will be zero, automatically No records found will apply